stages/test/test_bootupd.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python3
+
+import os.path
+from unittest.mock import call, patch
+
+import pytest
+
+import osbuild.meta
+from osbuild.testutil.imports import import_module_from_path
+
+
+@pytest.mark.parametrize("test_data,expected_err", [
+    # bad
+    ({"deployment": "must-be-object"}, "'must-be-object' is not of type 'object'"),
+    ({"deployment": {"osname": "some-os"}}, "'ref' is a required property"),
+    ({"deployment": {"ref": "some-ref"}}, "'osname' is a required property"),
+    ({"deployment": {"osname": "some-os", "ref": "some-ref", "serial": "must-be-number"}},
+     "'must-be-number' is not of type 'number'"),
+    ({"random": "propertey"}, "Additional properties are not allowed"),
```

Comment:
```suggestion
    ({"random": "property"}, "Additional properties are not allowed"),
```

mispelling here intentional?

Reply:
silly me, typo indeed